src/librustc_typeck/diagnostics.rs

```diff
@@ -1250,6 +1363,44 @@ information see the [opt-in builtin traits RFC](https://github.com/rust-lang/
 rfcs/blob/master/text/0019-opt-in-builtin-traits.md).
 "##,
 
+E0195: r##"
+Your method's lifetime parameters do not match the trait declaration.
+Erroneous code example:
+
+```
+trait Trait {
+    fn t<'a,'b:'a>(x: &'a str, y: &'b str);
+}
+
+struct Foo;
+
+impl Trait for Foo {
+    fn t<'a,'b>(x: &'a str, y: &'b str) { // error: lifetime parameters
+                                          //        or bounds on method `t`
+                                          //        do not match the trait
+                                          //        declaration
+    }
+}
+```
+
+The 'b lifetime constraints for `t` implementation does not match the
```

Comment:
- Backticks around `'b`
 - "The lifetime constraint `'b` for the implementation of `t()` does not match the trait declaration"
 - replacing `t` with `method` might be better